### ISSUE TITLE
[Feature]: Custom KDiffusion Scheduler

### ISSUE THREAD

**Issue by KohakuBlueleaf** [vladmandic/automatic#1242]
### Feature description

Based on https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/10621
and pr at https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/10649

I think it is worth to add this feature into this repo too.
But the problem is, I don't know if @vladmandic you want the settings one or something more like my original pr https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/10634

And if you want, I can add something like export the plot of the noise schedule. (I think this is doable?)

**Comment by vladmandic** [comment]
yes, i think that behavior like hires fix would be perfect - show options when checkbox is selected.
and i don't mind messing with scheduler list
i'll try to do the same for unipc as its settings are right now global.

**Comment by vladmandic** [comment]
@KohakuBlueleaf btw, since you're interested in schedulers, take a look at #1268. i think notes raised in paper are very valid and this is pretty high-value item (also enabled via checkbox to have backwards compatibility) - what do you think?

**Comment by KohakuBlueleaf** [comment]
@vladmandic first, there is 3 problem
1, snr=0 means sigma = inf, which is impossible in k_diffusion, you can only get some super high sigma (which already be possible with custom scheduler)

2, you need to change the alpha schedule in models to make sure k_diffusion has correct conversion between sigma and timesteps

3, I already try 1,2 on normal sd model and it just broken all the things, basically max_sigma higher than 1000 will give you super bad result. Normally max_sigma = 100 is the upper bound.
I guess this may caused by the scaling method and numerical error in k_diffusion


Conclusions:
If you have fixes for alpha schedule (which need to hijack sd-model), custom scheduler can let you use snr≈0.
Since I don't think use it for non-finetuned model is a good idea, I will not want to make a method to let user manualy change the alpha schedule in the model ckpt.

And if the model already contain fixed/different schedule, just use 0.0 for sigma min/max and it will read the value from models' params

**Comment by vladmandic** [comment]
@KohakuBlueleaf just looping back on this one - i see that your pr is merged in a1111 dev branch.
would you like to create pr here as well or should i look at porting at some time in the future.

**Comment by KohakuBlueleaf** [comment]
@vladmandic I'm quite busy these days (until 7/5 I guess)
I think it is actually not hard to port?
you can consider just port by yourself

**Comment by vladmandic** [comment]
finally got around to this :)
